### PR TITLE
Clarify instructions around creating issues

### PR DESCRIPTION
**Pull request by lauracowen**
Re-created from #2348 so we can close #2348.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,19 +29,15 @@
...
 
-   * **GA release blog posts** (Open Liberty GA release announcements only)
+   * **GA and beta release blog posts** (Open Liberty GA and beta release announcements only)
 
-     Create an issue using the `Open Liberty release notes` issue template. Make sure to select each task in the issue as you complete it to show progress.
-
-   * **Beta release blog posts** (Open Liberty beta release announcements only)
-
-     Create a normal issue. In the title, make clear that it's for the beta release and which release version.
+     Create a new issue using the `Open Liberty release notes` issue template. Make sure to select each task in the issue as you complete it to show progress.
```
Comment (dmuelle) [inline README.md:40]: is this still correct now that the GA and beta release posts are at least partially automated through GitHub actions? I'm not sure of the extent of that automation or if it makes this step obsolete

Comment (lauracowen) [reply README.md:40]: Hmm, I don't know what the process is now. @mbroz2 should probably comment...

Comment (mbroz2) [reply README.md:40]: haha, I just had a conversation with Reece earlier this week that I need to update documentation for the release/beta blog posts.  It's on my to-do, but I've also been waiting for the dust to settle with regards to GHA so I don't need to continuously update the documentation.

Comment (mbroz2) [reply README.md:40]: in the meantime, don't worry about this section as I've been doing all the recent release posts.

**Review by dmuelle — APPROVED** [review]
LGTM

**Pull request merged**